### PR TITLE
adding block model

### PR DESCRIPTION
**Pull request by rjmarin**
1.- Resize the block  with a new size (origin/rx , origin/ry, origin/rz) dimension of  (nx +1) *(ny+1) * (nz +1).  in all items sum mass counter that depends of the origin block its almost a transpose matrix but it doesn`t
2.-  the function is not performing correctly because it have a few bugs in the code for example at the firsts lines when you nedd to took the lenght of this block is not correct and the vars ny, nz always took 0.
The reblock does a block with only one direccionary with data (mass: 1) and the others are empty

### CODE REVIEW

File: block_model.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  attr_reader :blocks
+
+  def initialize(blocks)
+    @blocks = blocks
+  end
+
+  def reblock(rx, ry, rz)
+    new_blocks = []
+    nx = @blocks.length / rx
+    ny = @blocks.length[0] / ry
```
Comment (rjmarin) [inline block_model.rb]: correct from : ny = @blocks[0].length / ry

File: block_model.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    nz = @blocks.length[0][0] / rz
```
Comment (rjmarin) [inline block_model.rb]: correct from:  nz = @blocks[0][0].length / rz

File: block_model.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    n_i = 0
+    n_j = 0
+    n_k = 0
+    for i in (0..@blocks.length - 1).step(rx)
+      for j in (0..@blocks[0].length - 1).step(ry)
+        for k in (0..@blocks[0][0].length - 1).step(rz)
+          n = {}
+          for ri in (i..rx-1)
+            for rj in (i..ry-1)
+              for rk in (i..rx-1)
```
Comment (rjmarin) [inline block_model.rb:32]: Is not necessary to have a lot of fors in this function are many forms to solution  this reblock with a few fors(min 3)

File: block_model.rb (hunk 2 above)
Comment (rjmarin) [inline block_model.rb:12]: we have an error because the function length is not an array

File: block_model.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
+class BlockModel
```
Comment (rjmarin) [inline block_model.rb]: It is preferable

Comment (rjmarin) [inline block_model.rb]: Its necesary to comment the code for others members to understand what are doing the function  and it is prefeerable to use a good var names also.

File: block_model.rb [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  ],
+  [
+    [{mass: 1}, {mass: 1}],
+    [{mass: 1}, {mass: 1}],
+  ]
+]
+
+bm = BlockModel.new(blocks)
+bm.reblock(2,2,2)
+puts bm.blocks
```
Comment (rjmarin) [inline block_model.rb:73]: this part of the code its prettier to have in other file because the file is only of the class

**Review by rjmarin — COMMENTED** [review]
It is necessary to comment the code for others members to understand what are doing the function and it is preferable to use a good var names also.

**Review by alejandroechev — APPROVED** [review]
Good suggestions for improvements

**Pull request merged**